packages/oauth/src/index.ts

```diff
@@ -137,6 +195,52 @@ export class InstallProvider {
     }
   }
 
+  /**
+   * checkTokenExpiry determines access tokens' eligibility for refresh.
+   *
+   * The return value is an Array of expired or soon-to-expire access tokens.
+   */
+  private checkTokenExpiry(authResult: AuthorizeResult, currentUTCSec: number): string[] {
```

Comment:
I think `check` is generally a bit unclear naming. How about naming the method using `detect` or `extract`? The method's responsibility is to return expired/expiring tokens from the auth result.

```suggestion
  private detectExpiredOrExpiringTokens(authResult: AuthorizeResult, currentUTCSec: number): string[] {
```

Reply:
On another note, these new private methods are stateless (=they don't access the internal state of this object and probably they should not). You may want to place them at top level. You can call them without `self.`.

Reply:
Both great suggestions, I'll make the changes!

Reply:
When I went to actually make this change, it became apparent that the second function _does_ rely on the instance (`client`, `clientId` and `clientSecret`). If there is a way to do as you've indicated that I'm not seeing, happy to hear it and adjust, but for the time being, I'll just make the name change and move the detection function to the top level.